autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/sync/CredentialsSyncDataPersister.kt

```diff
@@ -0,0 +1,141 @@
+import com.duckduckgo.appbuildconfig.api.isInternalBuild
+import com.duckduckgo.autofill.sync.CredentialsSyncDataPersister.Adapters.Companion.updatesAdapter
+import com.duckduckgo.autofill.sync.persister.CredentialsMergeStrategy
+import com.duckduckgo.di.DaggerMap
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.sync.api.engine.SyncChangesResponse
+import com.duckduckgo.sync.api.engine.SyncMergeResult
+import com.duckduckgo.sync.api.engine.SyncMergeResult.Success
+import com.duckduckgo.sync.api.engine.SyncableDataPersister
+import com.duckduckgo.sync.api.engine.SyncableDataPersister.SyncConflictResolution
+import com.duckduckgo.sync.api.engine.SyncableType.CREDENTIALS
+import com.squareup.anvil.annotations.ContributesMultibinding
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import javax.inject.*
+import timber.log.Timber
+
+@ContributesMultibinding(scope = AppScope::class, boundType = SyncableDataPersister::class)
+class CredentialsSyncDataPersister @Inject constructor(
+    private val credentialsSyncMetadata: CredentialsSyncMetadata,
+    private val credentialsSyncStore: CredentialsSyncStore,
+    private val strategies: DaggerMap<SyncConflictResolution, CredentialsMergeStrategy>,
+    private val appBuildConfig: AppBuildConfig,
+) : SyncableDataPersister {
+    override fun persist(
+        changes: SyncChangesResponse,
+        conflictResolution: SyncConflictResolution,
+    ): SyncMergeResult<Boolean> {
+        if (appBuildConfig.isInternalBuild()) checkMainThread()
+
+        return if (changes.type == CREDENTIALS) {
+            Timber.d("Sync-autofill-Persist: received remote changes ${changes.jsonString}")
+            Timber.d("Sync-autofill-Persist: received remote changes, merging with resolution $conflictResolution")
+            val result = process(changes, conflictResolution)
+            Timber.d("Sync-autofill-Persist: merging credentials finished with $result")
+            return result
+        } else {
+            Timber.d("Sync-autofill-Persist: no credentials to merge")
+            Success(false)
+        }
+    }
+
+    private fun process(
+        changes: SyncChangesResponse,
+        conflictResolution: SyncConflictResolution,
+    ): SyncMergeResult<Boolean> {
+        val response = kotlin.runCatching {
+            updatesAdapter.fromJson(changes.jsonString)!!
+        }.getOrElse {
+            Timber.d("Sync-autofill-Persist: failed to parse remote changes")
+            return SyncMergeResult.Error(reason = "Error parsing credentials ${it.message}")
+        }
+
+        val result = processEntries(response.credentials, conflictResolution)
+
+        if (result is Success) {
+            pruneDeletedObjects(response.credentials.last_modified)
+        }
+
+        return result
+    }
+
+    private fun processEntries(
+        crendentials: CrendentialsSyncEntries,
```

Comment:
typo: 
`crendentials` used in many places in this class instead of `credentials`